playbooks/roles/cluster/tasks/workflow.yml

```diff
@@ -0,0 +1,150 @@
+---
+- name: Configure Openshift authenticate job template
+  tower_job_template:
+    name: "{{ cluster_job_template_openshift_auth_name }}"
+    description: "{{ cluster_job_template_openshift_auth_desc }}"
+    job_type: "{{ cluster_job_template_openshift_auth_type }}"
+    playbook: "{{ cluster_job_template_openshift_auth_playbook }}"
+    project: "{{ cluster_job_template_openshift_auth_project }}"
+    credential: "{{ tower_credential_bundle_default_name }}"
+    extra_vars_path: "roles/cluster/files/openshift_tower_auth.yml"
+    state: present
+    inventory: "Provisioning_inventory"
+  register: cluster_create_openshift_auth_out
+  until: cluster_create_openshift_auth_out is succeeded
+  retries: 10
+  delay: 5
+
+- name: Configure generate inventory job template
+  tower_job_template:
+    name: "{{ cluster_job_template_generate_inventory_name }}"
+    description: "{{ cluster_job_template_generate_inventory_desc }}"
+    job_type: "{{ cluster_job_template_generate_inventory_type }}"
+    playbook: "{{ cluster_job_template_generate_inventory_playbook }}"
+    project: "{{ cluster_job_template_generate_inventory_project }}"
+    credential: "{{ tower_credential_bundle_default_name }}"
+    state: present
+    inventory: "Provisioning_inventory"
+  register: cluster_create_generate_inventory_out
+  until: cluster_create_generate_inventory_out is succeeded
+  retries: 10
+  delay: 5
+
+- name: Add Custom Credential
+  command: tower-cli credential create --name "le_privatekey" --credential-type "le_privatekey" --inputs "{'le_key':'{{ letsencrypt_private_key }}'}" --organization "{{ tower_organization }}"
+  ignore_errors: true
+
+- name: Associate additional credentials
```

Comment:
Looks like there are a number of tasks named "Associate additional credentials". Could we change each of these so that they are unique? Was debugging which one failed and had to go through the order of the code to identify which one it was.

Reply:
I've added variables to all those tasks to make it clear what credential is being added to what job.